csmaps_utils.sh

```diff
@@ -0,0 +1,162 @@
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -o nounset
+
+OC=oc
+YQ=yq
+CM_NAME="mock-cs-maps"
+CONTROL_NS="cs-control"
+
+# update_cs_maps Updates the common-service-maps with the given yaml. Note that
+# the given yaml should have the right indentation/padding, minimum 2 spaces per
+# line. If there are multiple lines in the yaml, ensure that each line has
+# correct indentation.
+function update_cs_maps() {
+    local yaml=$1
+
+    local object="$(
+        cat <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "$CM_NAME"
+  namespace: kube-public
+data:
+  common-service-maps.yaml: |
+${yaml}
+EOF
+)"
+    echo "$object" | oc apply -f -
+}
+
+# create_empty_csmaps Creates a new common-service-maps configmap and inserts
+# an empty common-service-maps.yaml field.
+#
+# If the common-service-maps already exists, then will error
+function create_empty_csmaps() {
+    title " Creating empty common-service-maps configmap "
+    local isExists=$("${OC}" get configmap --ignore-not-found -n kube-public "$CM_NAME")
+    if [ ! -z "$isExists" ]; then
+        info "The $CM_NAME already exists, skipping"
+        return
+    fi
+    update_cs_maps ""
+    success "Empty common-service-maps configmap created in kube-public namespace"
+}
+
+# insert_control_ns Insert the controlNamespace field into the configmap if it
+# does not exist
+function insert_control_ns() {
+    local current_yaml=$("${OC}" get -n kube-public cm "$CM_NAME" -o yaml | "${YQ}" '.data.["common-service-maps.yaml"]')
+
+    current=$(echo "$current_yaml" | "${YQ}" .'controlNamespace')
+    if [[ "$current" != "$CONTROL_NS" && "$current" != "" && "$current" != "null" ]]; then
+        error "The controlNamespace field in common-service-maps is already set to: $current, and cannot be changed"
+    fi
+
+    local updated_yaml=$(echo "$current_yaml" | "${YQ}" '.controlNamespace = "'$CONTROL_NS'"')
+    local padded_yaml=$(echo "$updated_yaml" | awk '$0="    "$0')
+    update_cs_maps "$padded_yaml"
+}
+
+# read_tenant_from_csmaps Gets the list in requested-from-namespace for a given
+# map_to_cs_ns and prints it out. If map_to_cs_ns does not exist, then output is
+# empty
+function read_tenant_from_csmaps() {
+    local map_to_cs_ns=$1
+    local current_yaml=$("${OC}" get -n kube-public cm "$CM_NAME" -o yaml | "${YQ}" '.data.["common-service-maps.yaml"]')
+    local tenant_ns_list=$(echo "$current_yaml" | "${YQ}" eval '.namespaceMapping[] | select(.map-to-common-service-namespace == "'${map_to_cs_ns}'").requested-from-namespace' | awk '{ print $2 }')
```

Comment:
you might need to truncate newline characters here otherwise the list is passed on like:  "ns1\nns2\nns3 ...etc". You can use `| tr '\n' ' '` after the awk command

Reply:
`tenant_ns_list` does end up being newline delimited here. I don't think whether it's newline or space delimited impacts anything.